src/main/scala/org/broadinstitute/dsde/workbench/avram/util/RestClient.scala

```diff
@@ -0,0 +1,27 @@
+package org.broadinstitute.dsde.workbench.avram.util
+
+import java.net.URI
+import java.util.logging.Logger
+
+import com.softwaremill.sttp.{HttpURLConnectionBackend, Request, Uri, sttp}
+
+/**
+  * Helpful stuff for making rest calls using sttp (https://github.com/softwaremill/sttp).
+  */
+trait RestClient {
+  private val log: Logger = Logger.getLogger(getClass.getName)
+
+  implicit val sttpBackend = HttpURLConnectionBackend() // TODO: consider refactoring to allow use of SttpBackendStub in tests
```

Comment:
Is it ok to have a shared backend for multiple request threads? In other words should this be a `def`?

Reply:
Is there just one instance? My java background makes me think of a trait as a fancy interface, where nothing actually exists until an object or class instance extends the trait. Further, I can't access `RestClient.sttpBackend` directly as I would expect to be able to if it was effectively a singleton.

I'd have to run some more experiments to prove whether or not scala is retaining a single value and giving every object a reference to it, but I hope it's not doing that. I hope my mental model isn't that far off. 😬

Reply:
I think it's just 1 instance because there's just 1 instance of `SamDao`, which extends this trait.

I think it's probably fine, I just like to call out cases where singletons are used concurrently since I've been burned by that before.

Reply:
Ah, you're right. I must be thinking of some other context (maybe Akka land?). I'll make sure HttpURLConnectionBackend is thread-safe.

Reply:
Update, I had looked into this before and concluded that it is thread-safe, but I still changed it to a `def` because I don't see any downside.